### ISSUE TITLE
Saving map reverts all changes to map structure that appeared before opening map editor

### ISSUE THREAD

**Issue by fluentpwn** [sirjuddington/SLADE#1449]
**SLADE Version**: latest commit build
**Operating System**: Linux
Sys.info(`inxi --system`):
```
System:
  Host: artix Kernel: 6.0.12-artix1-1 arch: x86_64 bits: 64 Desktop: dwm
    v: 6.4 Distro: Artix Linux
```

**Issue Details**:
I have 2 windows open - main window and the map editor.
If I change the script(that includes just creating the script) and compile it, and save with `Ctrl+S`, upon saving the map all the changes I made before (re)opening the map editor get reverted(before I see that there is also an `Invalid Map` error). If I reopen the map editor after saving files and then save the map, it still gives me an `Invalid Map` error but doesn't revert any changes I made(just because it probably can't?).
The only map format I checked is UDMF, but might as well be other formats.
NOTE: `Invalid Map` error occurs only the first time, then it does the same but there is no error.

So basically, reproducing steps:
1. Create a WAD(I checked only doom2 iwad but it's probably not the cause)
2. Create a map(as mentioned, checked UDMF format only)
3. Open map editor
4. Change any files&/lumps
5. Save the files
6. Save the map
7. !

Video: https://0x0.st/o59I.mkv

**Comment by sirjuddington** [comment]
You're not meant to edit a map's entries in the main editor while it is open in the map editor (there is a script window in the map editor if you need to edit scripts)

I thought the map entries were supposed to locked while open in the map editor, but that doesn't appear to be the case for some reason.

I've just committed a change so that map entries are locked when open in the map editor, so this situation shouldn't be able to happen (at least not as easily as in the video)
